### DISCUSSION TITLE
HTTP not able to get responses?

### DISCUSSION THREAD

**Discussion by Zesheir** [cc-tweaked/CC-Tweaked#1961] [Q&A]
Having trouble getting the HTTP api to work at all - when I attempt to http.get far as I can tell **any** url, it doesn't return any sort of response. Tried checking for an HTTP success/failure event using http.request, then listening for said events, following along with how the wiki does it. It ends up sitting endlessly, so either the event(s) aren't firing for some reason or it's just timing out trying to read the response. Printing the results of http.response just returns (true, nil), and I'm not entirely sure what to do with that or how to fix it. Attached is the code I'm using and the print I'm getting. I'm sure I'm being stupid _somewhere_, so if anyone knows what the hell is going on/what I'm doing wrong, that would be [great.]
![2024-08-30_13 37 51](https://github.com/user-attachments/assets/e37a4ed3-faf5-4eb4-8392-0d6a2d319f8d)
![2024-08-30_13 38 10](https://github.com/user-attachments/assets/ebca1c3d-50ab-419c-b89b-82296cd877d6)

**Comment by Zesheir** [comment]
HTTP is enabled in config as well, forgot to mention.

**Comment by SquidDev** [comment]
The `http_success` event is only queued if the HTTP request succeeded, so you'd need to listen to `http_failure` events too (the docs probably should just mention that).

If you wrap `http.get` in an `assert`, it should print out the error that occurred (`local request = assert(http.get("https://example.tweaked.cc"))`).

Does anything appear in the logs? Typically this is caused by an TLS certificate error (see #1719), which should log an error.

**Reply by Zesheir** [reply]
Sorry if I wasn't clear, but I checked for both success and failure events with the same result (which makes no sense to me, i would expect to get one or the other but it just seems to get stuck. Ran an assert like you suggested, gave a "could not connect" error. Nothing I can find in logs or otherwise, just that. 
![2024-08-30_14 38 40](https://github.com/user-attachments/assets/1109ea8b-6a89-404f-b4d2-f2d32e26c984)

**Reply by SquidDev** [reply]
Yeah, that's very odd. `http.get` just listens to those events behind the scenes, so it definitely *should* be queuing them.

It might be worth checking your firewall options, just to see if Java is being blocked from accessing the network. Though then I'd expect your Minecraft skin wouldn't work (nor would you be able to play on servers), so that's probably not the case.

Otherwise, I'm afraid I don't have any idea at all what's going on here!

**Reply by Zesheir** [reply]
Well shit. I did check that, I've had weird firewall interactions before but only when hosting servers, and currently everything works fine. I'll probably keep fiddling with it, try on a different computer, as well as a different network maybe, see if I can't get it to work. Thanks though.
